pyproject.toml

```diff
@@ -39,7 +39,7 @@ pydantic = "^2.3.0"
 psutil = "^5.9.0"
 dacite = "^1.6.0"
 adlfs = "^2023.9.0"
-neo4j = "^4.4.3"
```

Comment:
this pushes users to use neo4j v5. Are we ok with this? that means no more v4

Reply:
@as51340 intent was to test if this works on a version bigger than v4 and use v5; we are not going to push users to use that version.  PR auto-added you for a review since I need to set it to read because of the CI/CD test.